scipy/stats/_hypotests.py

```diff
@@ -130,3 +131,133 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
     p = chi2.sf(w, r)
 
     return Epps_Singleton_2sampResult(w, p)
+
+
+def poisson_etest(k1, k2, n1, n2, diff=0, alternative='two-sided'):
+    """
+    Calculate E-test for the mean difference of two samples
+
+    The test requires two samples coming from poisson distribution. This is a
+    right-sided and two-sided test.
+
+    Parameters
+    ----------
+    k1, k2 : int or float
+        Count from the first and second samples respectively
+    n1, n2 : int or float
+        Sample size of each sample
+    diff : int of float, optional
+        The difference of mean between two samples under null hypothesis
+    alternative : {'two-sided', 'greater'}, optional
+        Whether to get p-value for two-sided hypothesis or right sided hypothesis.
+        Right-sided test that mean from sample one is greater than sample two
+
+    Returns
+    -------
+    pvalue : float
+        The associated p-value based on estimated p-value of the standardized
+        difference.
+
+    Notes
+    -----
+    The Poisson distribution is commonly used to model many processes such as
+    transactions per user. The usual test to compare difference between two
+    means of Poisson samples is C-test, based on conditional distribution.
```

Comment:
add ref to Przyborowski and Wilenski (1940)

Reply:
see below example of how references are listed